openmicroscopy/openmicroscopy
# Size limit for download as jpg

See https://trac.openmicroscopy.org/ome/ticket/12982

This limits the download as jpeg / png / tiff to images of 12k * 12k or smaller.
This cut-off was determined by trying to download images of various sizes.
Above 2147483648 pixels (46k * 46k) we get 
```
java.lang.NegativeArraySizeException: null
    at omeis.providers.re.RGBIntBuffer.<init>(RGBIntBuffer.java:62) ~[common.jar:na]
```
For 15k images we get
```
java.lang.OutOfMemoryError: Java heap space
    at omeis.providers.re.RGBIntBuffer.<init>(RGBIntBuffer.java:62) ~[common.jar:na]
    at omeis.providers.re.RenderingStrategy.getIntBuffer(RenderingStrategy.java:186) ~[rendering.jar:na]
    at omeis.providers.re.HSBStrategy.renderAsPackedInt(HSBStrategy.java:275) ~[rendering.jar:na]
```
but 12k images download OK, but quite slowly.

To test:
 - Select an image smaller than 12k * 12k or equivalent, and try to download as jpg/png/tiff (should work for 'big' images as well as regular images.
 - Select an image bigger than 12k * 12k and check that the download as jpg/png/tiff options are disabled with a suitable tooltip.
 - Select multiple images. If any single image is too big, the download as jpg/png/tiff options should be disabled. Otherwise, download as a zip should work OK.
